lib/ansible/modules/cloud/amazon/_ec2_vpc.py

```diff
@@ -713,30 +721,32 @@ def terminate_vpc(module, vpc_conn, vpc_id=None, cidr=None):
 
 def main():
     argument_spec = ec2_argument_spec()
-    argument_spec.update(dict(
-        cidr_block = dict(),
-        instance_tenancy = dict(choices=['default', 'dedicated'], default='default'),
-        wait = dict(type='bool', default=False),
-        wait_timeout = dict(default=300),
-        dns_support = dict(type='bool', default=True),
-        dns_hostnames = dict(type='bool', default=True),
-        subnets = dict(type='list'),
-        vpc_id = dict(),
-        internet_gateway = dict(type='bool', default=False),
-        resource_tags = dict(type='dict', required=True),
-        route_tables = dict(type='list'),
-        state = dict(choices=['present', 'absent'], default='present'),
-    )
+    argument_spec.update(
+        dict(
+            cidr_block=dict(),
+            instance_tenancy=dict(choices=['default', 'dedicated'], default='default'),
+            wait=dict(type='bool', default=False),
+            wait_timeout=dict(default=300),
+            dns_support=dict(type='bool', default=True),
+            dns_hostnames=dict(type='bool', default=True),
+            subnets=dict(type='list'),
+            vpc_id=dict(),
+            internet_gateway=dict(type='bool', default=False),
+            resource_tags=dict(type='dict'),
+            route_tables=dict(type='list'),
+            state=dict(choices=['present', 'absent'], default='present'),
+        )
     )
 
     module = AnsibleModule(
         argument_spec=argument_spec,
+        required_one_of=[['vpc_id', 'resource_tags']],
```

Comment:
Would a required_if for state absent be better? It's weird to require one of these for creation if you might just plan to use vpc_id to then delete it.